### PR TITLE
Add `CloudSeedProvider`

### PR DESCRIPTION
**Pull request by jackwaudby**

### CODE REVIEW

**Comment by neo-technology-commit-status-publisher** [comment]
This PR includes documentation updates
View the updated docs at https://neo4j-docs-operations-1906.surge.sh

Updated pages:
- [Managing databases in a cluster](https://neo4j-docs-operations-1906.surge.sh/operations-manual/5/clustering/databases)

**Comment by neo-technology-commit-status-publisher** [comment]
This PR includes documentation updates
View the updated docs at https://neo4j-docs-operations-1906.surge.sh

Updated pages:
- [Managing databases in a cluster](https://neo4j-docs-operations-1906.surge.sh/operations-manual/5/clustering/databases)

File: modules/ROOT/pages/clustering/databases.adoc [hunk 1]
```diff
@@ -491,25 +495,68 @@ Each configuration value is specified as a name followed by `=` and the value, a
...
+
+The `CloudSeedProvider` supports:
+
+** `s3:`
+** `gs:`
+** `azb:`
+
+[.tabbed-example]
+=====
+[role=include-with-AWS-S3 label--new-5.19]
```
Comment (NataliaIvakina) [inline modules/ROOT/pages/clustering/databases.adoc:519]: I think it was introduced in 5.1.

File: modules/ROOT/pages/clustering/databases.adoc [hunk 2]
```diff
@@ -438,27 +438,13 @@ The seed can be either a backup or a dump from an existing database.
 The sources of seeds are called _seed providers_.
 
 The mechanism is pluggable, allowing new sources of seeds to be supported (see link:https://www.neo4j.com/docs/java-reference/current/extending-neo4j/project-setup/#extending-neo4j-plugin-seed-provider[Java Reference -> Implement custom seed providers] for more information).
-The product has built-in support for seed from a mounted file system (file), FTP server, HTTP/HTTPS server and Amazon S3.
+The product has built-in support for seed from a mounted file system (file), FTP server, HTTP/HTTPS server, Google Cloud Storage, Azure Cloud Storage, and Amazon S3.
```
Comment (NataliaIvakina) [inline modules/ROOT/pages/clustering/databases.adoc:441]:
```suggestion
The product has built-in support for seed from a mounted file system (file), FTP server, HTTP/HTTPS server, Amazon S3, Google Cloud Storage (from Neo4j 5.25), and Azure Cloud Storage (from Neo4j 5.25).
```

File: modules/ROOT/pages/clustering/databases.adoc [hunk 3]
```diff
@@ -491,25 +495,68 @@ Each configuration value is specified as a name followed by `=` and the value, a
...
+
+. Create database from `myBackup.backup`. 
++
+[source,shell, role="nocopy"]
+----
+CREATE DATABASE foo OPTIONS { existingData: 'use', seedURI: 's3:/myBucket/myBackup.backup' }
+----
+
+======
+[role=include-with-Google-cloud-storage label--new-5.24]
```
Comment (NataliaIvakina) [inline modules/ROOT/pages/clustering/databases.adoc:534]:
```suggestion
[role=include-with-Google-cloud-storage label--new-5.25]
```

File: modules/ROOT/pages/clustering/databases.adoc [hunk 4]
```diff
@@ -491,25 +495,68 @@ Each configuration value is specified as a name followed by `=` and the value, a
...
+include::partial$/gcs-credentials.adoc[]
+
+. Create database from `myBackup.backup`. 
++
+[source,shell]
+----
+CREATE DATABASE foo OPTIONS { existingData: 'use', seedURI: 'gs:/myBucket/myBackup.backup' }
+----
+======
+[role=include-with-Azure-cloud-storage label--new-5.24]
```
Comment (NataliaIvakina) [inline modules/ROOT/pages/clustering/databases.adoc:546]:
```suggestion
[role=include-with-Azure-cloud-storage label--new-5.25]
```

File: modules/ROOT/pages/clustering/databases.adoc [hunk 5]
```diff
@@ -481,7 +467,25 @@ neo4j@neo4j> SHOW DATABASES;
...
+** `http:`
+** `https:`
+
+The `S3SeedProvider` supports:
+
+** `s3:`
+
+[NOTE]
+====
+Neo4j 5 comes bundled with necessary libraries for AWS S3 connectivity and thus if using  `S3SeedProvider` then `aws cli` is not required, but can be used with the `CloudSeedProvider`.
```
Comment (NataliaIvakina) [inline modules/ROOT/pages/clustering/databases.adoc:485]:
```suggestion
Neo4j 5 comes bundled with necessary libraries for AWS S3 connectivity.
Therefore, if you use `S3SeedProvider`,`aws cli` is not required but can be used with the `CloudSeedProvider`.
```

File: modules/ROOT/pages/clustering/databases.adoc (hunk 1 above)
Comment (jackwaudby) [reply modules/ROOT/pages/clustering/databases.adoc:520]: Yes I think `S3SeedProvider` was, but this refers to CloudSeedProvider's s3 support so should be 5.25 - will change

**Comment by neo-technology-commit-status-publisher** [comment]
This PR includes documentation updates
View the updated docs at https://neo4j-docs-operations-1906.surge.sh

Updated pages:
- [Managing databases in a cluster](https://neo4j-docs-operations-1906.surge.sh/operations-manual/5/clustering/databases)

**Review by NataliaIvakina — APPROVED** [review]
Looks good! Thank you @jackwaudby !

**Pull request merged**